examples/training/classification/imagenet/basic_training.py

```diff
@@ -182,49 +206,98 @@ def augment(img, label):
         **eval(FLAGS.model_kwargs),
     )
 
-
 """
-Next, we pick an optimizer. Here we use Adam with a constant learning rate.
-Note that learning rate will decrease over time due to the ReduceLROnPlateau callback.
+Optional LR schedule with cosine decay instead of ReduceLROnPlateau
+TODO: Replace with Core Keras LRWarmup when it's released. This is a temporary solution.
 """
+def lr_warmup_cosine_decay(global_step,
+                           warmup_steps,
+                           hold=0,
+                           total_steps=0,
+                           start_lr=0.0,
+                           target_lr=1e-3):
+    # Cosine decay
+    # There is no tf.pi :(
```

Comment:
you can use np.pi?

Reply:
Yup, originally did, but then I saw no non-TF imports and thought that it might be discouraged 😬
Thinking back, pretty naive on my end. Pushing an update for this

Reply:
Done

Reply:
> Yup, originally did, but then I saw no non-TF imports and thought that it might be discouraged grimacing
You can also have
```python
import math as m
pi = tf.constant(m.pi)
```